packages/analytics/client/src/analytics_client/analytics_client.ts

```diff
@@ -360,3 +360,39 @@ export class AnalyticsClient implements IAnalyticsClient {
       });
   }
 }
+
+/** security paths that contain user data */
+const SECURITY_PATHS = ['hosts/name', 'users/name', 'network/ip'];
+// this indicates a user query
+const queryMarker = `?_g=`;
+const fixUrl = (url: string) => {
+  if (url.includes(queryMarker)) {
+    url = url.substring(0, url.indexOf(queryMarker) + queryMarker.length);
+  }
+  const matchedKnownPiiData = SECURITY_PATHS.find((path) => url.includes(path));
+
+  if (matchedKnownPiiData) {
+    const res = url.split(matchedKnownPiiData);
+    return res[0] + matchedKnownPiiData + res[1].replace(/^\/[^\/\r\n]+(?=\/)/, '/MASKED');
```

Comment:
nit: using split + regexp looks kind of odd to me... I wonder if we should replace the regexp with another split by `/` and drop the first element (AFAIK, that's essentially what the regexp is doing).